lib/container/HoistContainerReferencesPlugin.js

```diff
@@ -0,0 +1,205 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Zackary Jackson @ScriptedAlchemy
+*/
+
+"use strict";
+
+const { AsyncDependenciesBlock, ExternalModule } = require("webpack");
+const ModuleFederationPlugin = require("./ModuleFederationPlugin");
+
+const PLUGIN_NAME = "HoistContainerReferences";
+
+/**
+ * This class is used to hoist container references in the code.
+ */
+class HoistContainerReferences {
```

Comment:
in federation v2, we also call hook to add federation v2 runtime as addInclude, hoist container refs also hoists those referenced modules for that dependency as well so that we can ensure runtime plugins etc that user add, plus our own code is also eager regardless of chunk split rules etc. 
https://github.com/module-federation/core/blob/main/packages/enhanced/src/lib/container/ContainerPlugin.ts

Reference hoist ensures all external modules, remote modules, and v2 runtime is all in the runtime chunks and not in come other chunk lower in the tree.